src/internal/analytics-metadata/__tests__/labels-utils.test.tsx

```diff
@@ -225,6 +225,43 @@ describe('processLabel', () => {
     expect(processLabel(target, { selector: '.outer-class', root: 'body' })).toEqual('label outside of the component');
   });
 
+  test('respects the rootClassName property', () => {
+    const { container } = render(
+      <div className="root-class">
+        <div className="label-class">outer label</div>
+        <div id="target">
+          <div className="label-class">inner label</div>
+        </div>
+      </div>
+    );
+    const target = container.querySelector('#target') as HTMLElement;
+    expect(processLabel(target, { selector: '.label-class', rootClassName: 'root-class' })).toEqual('outer label');
+  });
+  test('rootClassName prevails over root property', () => {
+    const { container } = render(
+      <>
+        <div className="root-class">
+          <div className="label-class">root class label</div>
+          <div {...getAnalyticsMetadataAttribute({ component: { name: 'ComponentName' } })}>
+            <div className="label-class">component label</div>
+            <div id="target">
+              <div className="label-class">inner label</div>
+            </div>
+          </div>
+        </div>
+        <div className="outer-class">label outside of the component</div>
+      </>
+    );
+    const target = container.querySelector('#target') as HTMLElement;
+    expect(processLabel(target, { selector: '.label-class', root: 'self', rootClassName: 'root-class' })).toEqual(
+      'root class label'
```

Comment:
Why does it prevail over root: 'self'? My expectation from the PR description was that it should only affect the root: 'component'.

Reply:
it is a more granular description of what you want the root to be, so I think it should win. And it is independent of the `root=component`. it solves similar problems, but that does not mean that `rootClassName` will always be the root of the component

Reply:
Is there any difference between these two definitions: `{ root: 'self', rootClassName: 'root-class' }` and `{ root: 'component', rootClassName: 'root-class' }`?

Reply:
No, there is no difference. When you use `rootSelector` the `root` property is ignored.

Reply:
Should we then make root optional and not provide it here to avoid confusion?

Reply:
The root is already optional. We need to provide it here because the purpose of the test is test that the `rootSelector` prevails

Reply:
Ah, got it 👍